service/src/test/java/bio/terra/workspace/service/resource/controlled/cloud/azure/vm/EnableVmLoggingStepTest.java

```diff
@@ -0,0 +1,165 @@
+package bio.terra.workspace.service.resource.controlled.cloud.azure.vm;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.*;
+
+import bio.terra.common.iam.BearerToken;
+import bio.terra.landingzone.library.landingzones.deployment.ResourcePurpose;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.StepResult;
+import bio.terra.workspace.amalgam.landingzone.azure.LandingZoneApiDispatch;
+import bio.terra.workspace.app.configuration.external.AzureConfiguration;
+import bio.terra.workspace.common.BaseAzureUnitTest;
+import bio.terra.workspace.common.exception.AzureManagementExceptionUtils;
+import bio.terra.workspace.common.fixtures.WorkspaceFixtures;
+import bio.terra.workspace.generated.model.ApiAzureLandingZoneDeployedResource;
+import bio.terra.workspace.generated.model.ApiAzureLandingZoneResourcesList;
+import bio.terra.workspace.service.crl.CrlService;
+import bio.terra.workspace.service.iam.SamService;
+import bio.terra.workspace.service.workspace.WorkspaceService;
+import bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys;
+import bio.terra.workspace.service.workspace.model.AzureCloudContext;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.management.exception.ManagementException;
+import com.azure.resourcemanager.compute.ComputeManager;
+import com.azure.resourcemanager.compute.models.VirtualMachine;
+import com.azure.resourcemanager.compute.models.VirtualMachineExtension;
+import com.azure.resourcemanager.compute.models.VirtualMachines;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.MockedStatic;
+
+class EnableVmLoggingStepTest extends BaseAzureUnitTest {
+
+  /* these are all used in the constructor to EnableVmLoggingStep */
+  @Mock private AzureConfiguration azureConfig;
+  @Mock private CrlService crlService;
+  @Mock private ControlledAzureVmResource resource;
+  @Mock private LandingZoneApiDispatch landingZoneApiDispatch;
+  @Mock private SamService samService;
+  @Mock private WorkspaceService mockWorkspaceService;
+
+  @Mock private HttpResponse mockHttpResponse;
+
+  @Mock private FlightContext flightContext;
+
+  private BearerToken bearerToken;
+  private VirtualMachine.Update updateMachineMock;
+
+  @BeforeEach
+  void setup() {
```

Comment:
Out of scope for this review, but food for thought: I'm starting to push more on separating this sort of cloud business logic out of the "raw" stairway steps into helpers/utils/services/whatever to ease testing. 

Ideally we'd be able to isolate the azure interaction concerns and expose a saner interface to the stairway orchestration so we can exercise them separately in tests (or integration tests) and don't have to go into the heavyweight mocking as much.

Reply:
😬 ... this was definitely the pain point of getting this in. I don't mind moving it over there to help future people in this repo! ... here, hold my (insert-beverage-here)

Reply:
I was hoping to do this before break ... but got interrupted by other things. 
I am going to:
- get this merged, and
- follow-up with you to figure out how to do this (refactor)

Reply:
I don't have enough thumbs to express my support for this